adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m

```diff
@@ -0,0 +1,86 @@
+//
+//  GADPangleRTBBannerRenderer.m
+//  PangleAdapter
+//
+//  Created by bytedance on 2022/1/11.
+//
+
+#import "GADPangleRTBBannerRenderer.h"
+#import "GADMediationAdapterPangleConstants.h"
+#import <BUAdSDK/BUAdSDK.h>
+#import "GADMAdapterPangleUtils.h"
+
+@interface GADPangleRTBBannerRenderer()<BUNativeExpressBannerViewDelegate>
+
+@end
+
+@implementation GADPangleRTBBannerRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationBannerLoadCompletionHandler _loadCompletionHandler;
+    /// The Pangle banner ad.
+    BUNativeExpressBannerView *_nativeExpressBannerView;
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationBannerAdEventDelegate> _delegate;
+    /// The requested ad size.
+    CGSize _bannerSize;
+}
+
+- (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                     completionHandler:
+(nonnull GADMediationBannerLoadCompletionHandler)completionHandler {
+    _loadCompletionHandler = completionHandler;
+    NSString *slotId = adConfiguration.credentials.settings[GADMAdapterPanglePlacementID] ?: @"";
+    if (PangleIsEmptyString(slotId)) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorSlotIdNil, [NSString stringWithFormat:@"%@ cannot be nil.",GADMAdapterPanglePlacementID]);
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    _bannerSize = adConfiguration.adSize.size;
```

Comment:
We should verify Pangle supports the banner size in the request.

Reply:
Note that Pangle SDK Banner ad size can be adaptive，so no adjustment is needed here

Reply:
Would still prefer using `GADClosestValidSizeForAdSizes` and `GADAdSizeEqualToSize` to validate is the request banner size valid or not.

See this as an example:
https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Facebook/FacebookAdapter/GADFBBannerAd.m#L37